app/questionnaire/routing/when_rule_evaluator.py

```diff
@@ -33,11 +33,13 @@ def __post_init__(self) -> None:
             use_default_answer=True,
         )
 
-    def _evaluate(self, rule: dict[str, list]) -> Union[bool, Optional[date]]:
+    def _evaluate(
+        self, rule: dict[str, Union[list, tuple]]
+    ) -> Union[bool, Optional[date]]:
```

Comment:
Opening it up to list/tuple muddies the type check, I think we should update it to allow any `Sequence` then list/tuple would work fine.
```suggestion
    def _evaluate(self, rule: dict[str, Sequence]) -> Union[bool, Optional[date]]:
```